crypto/curve25519.go

```diff
@@ -123,6 +123,17 @@ func ed25519Verify(public ed25519PublicKey, data []byte, sig ed25519Signature) b
 	return result == 0
 }
 
+func ed25519VerifyBatchVerificationCompatibleVersion(public ed25519PublicKey, data []byte, sig ed25519Signature) bool {
```

Comment:
I think you should merge this with the ed25519Verify function, that way you can be sure to catch everything using crypto_sign_ed25519_verify_detached at the lowest level.

Reply:
I implemented it — caught some more signature verifications and wired them up to the consensus param for signatures in votes and compact certs: https://github.com/algoidan/go-algorand/pull/14/files